### ISSUE TITLE
Handle grafana queries directly

### ISSUE THREAD

**Issue by jeffpierce** [jeffpierce/cassabon#79]
It'd be nice to simply handle queries from Grafana directly rather than needing graphite or graphite-API as a middleman.

Let's look at how feasible this might be.

**Comment by mredivo** [comment]
What is the transform that Graphite does to requests, incoming and returning? Is the data very similar?

**Comment by jeffpierce** [comment]
It does a lot, based on the functions asked for.  We'd have to implement the functions listed here, at minimum:

http://graphite.readthedocs.org/en/latest/functions.html

**Comment by mredivo** [comment]
Hm, that is indeed a lot. Maybe the Unix philosophy of "do one thing well, and connect with pipes" applies here.

**Comment by jeffpierce** [comment]
Yep.  That's more of a pipe dream on my end of things, given that graphite-api is now the slowest part of the grafana -> graphite-api -> cassabon chain.

**Comment by utkarshcmu** [comment]
Some work is done in this repo - https://github.com/graphite-ng/graphite-api-experiment
